syntax = "proto3";
option go_package = "./;proto";

import "google/protobuf/empty.proto";

service User {
  rpc GetUserList(PageInfo) returns (UserListResponse);           // 查询用户列表
  rpc GetUserByMobile(MobileReqeust) returns (UserInfoResponse);  // 通过手机号查询用户
  rpc GetUserById(IdReqeust) returns (UserInfoResponse);          // 通过用户ID查询手机号
  rpc CreateUser(CreateUserInfo) returns (UserInfoResponse);      // 添加用户
  rpc UpdateUser(UpdateUserInfo) returns (google.protobuf.Empty); // 更新用户
}

message PageInfo {
  uint32 pn = 1;
  uint32 pSize = 2;
}
message MobileReqeust {
  string mobile = 1;
}
message IdReqeust {
  uint32 ID = 1;
}
message CreateUserInfo {
  string nickName = 1;
  string password = 2;
  string mobile = 3;

}
message UpdateUserInfo {
  int32 Id = 1;
  string nickName = 2;
  string gender = 3;
  uint64 birthday = 4;
}


message UserInfoResponse {
  int32 id = 1;
  string password = 2;
  string mobile = 3;
  string nickName = 4;
  string birthDay = 5;
  string gender = 6;
  int32 role = 7;
}

message UserListResponse {
  int32 total = 1;
  repeated UserInfoResponse data = 2;
}
